### PR TITLE
SLVSCODE-322 delete bindings when connection is deleted

### PR DESCRIPTION
**Pull request by sophio-japharidze-sonarsource**

### CODE REVIEW

File: src/settings.ts [hunk 1]
```diff
@@ -236,6 +237,15 @@ export class ConnectionSettingsService {
     return loadedConnection;
   }
 
+  async removeBindingsForConnection(connectionId: string) {
+    const bindingService = new BindingService(this.client, this);
```
Comment (sophio-japharidze-sonarsource) [inline src/settings.ts:241]: Initialisation is needed here (as opposed to passing the service to the constructor) to "avoid" circular dependency between the two services

File: src/extension.ts [hunk 2]
```diff
@@ -374,7 +374,12 @@ export function activate(context: VSCode.ExtensionContext) {
   context.subscriptions.push(
     VSCode.commands.registerCommand(
       Commands.REMOVE_CONNECTION,
-      (connection) => ConnectionSettingsService.instance.removeConnection(connection)
+      async (connection) => {
+        const connectionDeleted = await ConnectionSettingsService.instance.removeConnection(connection);
+        if(connectionDeleted){
```
Comment (kirill-knize-sonarsource) [inline src/extension.ts:379]: Formatting issue, spaces missing.

File: src/binding.ts [hunk 3]
```diff
@@ -61,11 +61,22 @@ export class BindingService {
...
+    const folder = workspaceFolderItem instanceof WorkspaceFolderItem ?
+                                workspaceFolderItem.uri : workspaceFolderItem.folder;
+    const config = VSCode.workspace.getConfiguration(SONARLINT_CATEGORY, folder);
     return config.update(BINDING_SETTINGS, undefined, VSCode.ConfigurationTarget.WorkspaceFolder);
   }
 
+  deleteBindingsForConnection(connection: Connection) {
+    const connectionId = connection.id || DEFAULT_CONNECTION_ID;
+    const allBindings = this.getAllBindings();
+    const bindingsForConnection : Map<string, BoundFolder[]> = allBindings.get(connectionId);
```
Comment (kirill-knize-sonarsource) [inline src/binding.ts:74]: Formatting issue, extra space.

File: src/binding.ts [hunk 4]
```diff
@@ -61,11 +61,22 @@ export class BindingService {
     return this.deleteBinding(binding);
   }
 
-  async deleteBinding(binding: WorkspaceFolderItem): Promise<void> {
-    const config = VSCode.workspace.getConfiguration(SONARLINT_CATEGORY, binding.uri);
+  async deleteBinding(workspaceFolderItem: WorkspaceFolderItem | BoundFolder): Promise<void> {
+    const folder = workspaceFolderItem instanceof WorkspaceFolderItem ?
+                                workspaceFolderItem.uri : workspaceFolderItem.folder;
```
Comment (kirill-knize-sonarsource) [inline src/binding.ts:66]: Formatting issue, indent is too big.

**Review by kirill-knize-sonarsource — APPROVED** [review]
Some formatting issues, else LGTM.

**Pull request merged**